### PR TITLE
homework 40 solution

### PR DESCRIPTION
**Pull request by tkhambir**

### CODE REVIEW

**Review by ilya449 — APPROVED** [review]

**Comment by ilya449** [comment]
Good job!

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -49,5 +51,15 @@ public static void main(String[] args) throws AuthenticationException {
...
+
+        ShoppingCartService shoppingCartService
+                = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+        UserService userService
+                = (UserService) injector.getInstance(UserService.class);
+        User userFromDb = userService.findByEmail("user1@gmail.com").get();
+        shoppingCartService.addSession(movieSession1, userFromDb);
+        System.out.println("Cart with tickets: " + shoppingCartService.getByUser(userFromDb));
+        shoppingCartService.clear(shoppingCartService.getByUser(userFromDb));
+        System.out.println("Empty cart: " + shoppingCartService.getByUser(userFromDb));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: can we save shopping cart to a local variable?

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -13,12 +14,17 @@
...
+
     @Override
     public User add(User user) {
         byte[] salt = HashUtil.getSalt();
         user.setSalt(salt);
         user.setPassword(HashUtil.hashPassword(user.getPassword(), salt));
-        return userDao.add(user);
+        User newUser = userDao.add(user);
+        shoppingCartService.registerNewShoppingCart(newUser);
+        return newUser;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: let's move this logic to `register()` method instead

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -18,7 +18,7 @@ public User add(User user) {
         byte[] salt = HashUtil.getSalt();
         user.setSalt(salt);
         user.setPassword(HashUtil.hashPassword(user.getPassword(), salt));
-        return userDao.add(user);
+        return user;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: let's return adding the user to db

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**